lgc/patch/FragColorExport.cpp

```diff
@@ -431,3 +484,331 @@ Value *FragColorExport::convertToInt(Value *value, bool signedness, Instruction
+  if (m_fragDepth) {
+    assert(builtInUsage.fragDepth);
+    (void(builtInUsage)); // unused
+    fragDepth = m_fragDepth;
+  }
+
+  if (m_fragStencilRef) {
+    assert(builtInUsage.fragStencilRef);
+    (void(builtInUsage)); // unused
+    channelMask |= 2;
+    fragStencilRef = m_fragStencilRef;
+  }
+
+  if (m_sampleMask) {
+    assert(builtInUsage.sampleMask);
+    (void(builtInUsage)); // unused
+    channelMask |= 4;
+    sampleMask = m_sampleMask;
+  }
+
+  Value *args[] = {
+      ConstantInt::get(Type::getInt32Ty(*m_context), EXP_TARGET_Z), // tgt
+      ConstantInt::get(Type::getInt32Ty(*m_context), channelMask),  // en
+      fragDepth,                                                    // src0
+      fragStencilRef,                                               // src1
+      sampleMask,                                                   // src2
+      undef,                                                        // src3
+      ConstantInt::get(Type::getInt1Ty(*m_context), false),         // done
+      ConstantInt::get(Type::getInt1Ty(*m_context), true)           // vm
+  };
+  return emitCall("llvm.amdgcn.exp.f32", Type::getVoidTy(*m_context), args, {}, insertPos);
+}
+
+// =====================================================================================================================
+// Generates a dummy export instruction.  Returns last export instruction that was generated.
+//
+// @param insertPos : All new instructions will be inserted before this position.
+CallInst *LowerFragColorExport::AddDummyExport(Instruction *insertPos) {
+  auto zero = ConstantFP::get(Type::getFloatTy(*m_context), 0.0);
+  auto undef = UndefValue::get(Type::getFloatTy(*m_context));
+  Value *args[] = {
+      ConstantInt::get(Type::getInt32Ty(*m_context), EXP_TARGET_MRT_0), // tgt
+      ConstantInt::get(Type::getInt32Ty(*m_context), 0x1),              // en
+      zero,                                                             // src0
+      undef,                                                            // src1
+      undef,                                                            // src2
+      undef,                                                            // src3
+      ConstantInt::get(Type::getInt1Ty(*m_context), false),             // done
+      ConstantInt::get(Type::getInt1Ty(*m_context), true)               // vm
+  };
+  return emitCall("llvm.amdgcn.exp.f32", Type::getVoidTy(*m_context), args, {}, insertPos);
+}
+
+// =====================================================================================================================
+// Sets the done flag on the given export instruction.
+//
+// @param [in/out] exportInst : The export instruction to be updated.
+void LowerFragColorExport::SetDoneFlag(CallInst *exportInst) {
+  if (exportInst == nullptr) {
+    return;
+  }
+
+  auto exportName = exportInst->getCalledFunction()->getName();
+  if (exportName == "llvm.amdgcn.exp.f32")
```

Comment:
This is an intrinsic, so you can use a Function method to get the intrinsic ID and compare on that.